src/Features/Core/Portable/Completion/CompletionItem.cs

```diff
@@ -211,6 +265,23 @@ public sealed class CompletionItem : IComparable<CompletionItem>
             string? displayTextSuffix = null,
             string? inlineDescription = null,
             bool isComplexTextEdit = false)
+        {
+            return CreateInternal(
+                displayText, filterText, sortText, properties != null ? properties.ToImmutableArray() : default, tags, rules, displayTextPrefix,
```

Comment:
Since we have the ImmutableDictionary here already, we should set `_propertiesAsImmutableDictionary` to it. Would save an allocation for users of this public API who access Properties later.

Reply:
I'm going back and forth on whether to change this. On the one hand, it *will* save an alloc if they later access the Properties immutable dictionary. On the other hand, it will pin this memory if they don't access it, and the (Try)GetProperty calls will be slower on small (most) property collections.

Preference?

Reply:
OK, I don't feel too strongly, so I went ahead and store the immutable dictionary if it's given.